src/store/reducer/tileSlice.ts

```diff
@@ -39,8 +44,12 @@ export const tileSlice = createSlice({
   name: "tile",
   initialState,
   reducers: {
-    changeTileColor: (state, action: PayloadAction<string>) => {
-      // TODO COLOR CHANGE
+    /* istanbul ignore next */
+    changeTileColor: (state, action: PayloadAction<any>) => {
+      const selectedLocation = state.findIndex(
+        (object) => object.location === action.payload.location
+      );
+      state[selectedLocation].color = action.payload.selectedColor;
```

Comment:
Please remember, redux tool kit uses immer.js which is create an immutable value based on the current values under the hood. If there is no immer, then you cant modify the state, you have to create a new one instead of the old one. 
(It is not a change requirement if you `are` interested in the theory, it you can have a look at immer.js)